src/components/FiltersBar/FiltersBar.tsx

```diff
@@ -0,0 +1,500 @@
+                    </Button>
+                    <Button size="small" onClick={handleApplyFilter}>
+                      Apply
+                    </Button>
+                  </>
+                }
+              >
+                <Button badge={licensesFilter?.length} variant="secondary" onClick={() => null}>
+                  License
+                </Button>
+              </PopoverDialog>
+              <PopoverDialog
+                content={
+                  <>
+                    <OtherFilterStyledText secondary variant="overhead">
+                      <OtherFilterStyledIcon />
+                      Exclude:
+                    </OtherFilterStyledText>
+                    {otherFiltersInputs}
+                  </>
+                }
+                footer={
+                  <>
+                    <Button
+                      onClick={clearOtherFilters}
+                      size="small"
+                      variant="secondary"
+                      disabled={!canClearOtherFilters}
+                    >
+                      Clear
+                    </Button>
+                    <Button size="small" onClick={handleApplyFilter}>
+                      Apply
+                    </Button>
+                  </>
+                }
+              >
+                <Button
+                  badge={(paidPromotionalMaterialFilter ? 1 : 0) + (matureContentRatingFilter ? 1 : 0)}
+                  variant="secondary"
+                  onClick={() => null}
+                >
+                  Other filters
+                </Button>
+              </PopoverDialog>
+            </FiltersInnerContainer>
+
+            {canClearAllFilters && (
+              <Button onClick={clearAllFilters} variant="tertiary" icon={<SvgGlyphClose />}>
+                Clear all
+              </Button>
+            )}
+          </FiltersContainer>
+        </CSSTransition>
+      )}
+    </>
+  )
+}
+
+const MobileFilterDialog: React.FC<{ title: string; content: React.ReactNode } & ActionDialogProps> = ({
+  title,
+  content,
+  ...actionDialogProps
+}) => {
+  return (
+    <ActionDialog {...actionDialogProps}>
+      {title && <StyledTitleText variant="h4">{title}</StyledTitleText>}
+      {content}
+    </ActionDialog>
+  )
+}
+
+export const useFiltersBar = () => {
+  const { id } = useParams()
+
+  // filters
+  const [selectedLanguage, setSelectedLanguage] = useState<string | null | undefined>('en')
+  const [sortVideosBy, setSortVideosBy] = useState<VideoOrderByInput>(VideoOrderByInput.CreatedAtDesc)
+  const [dateUploadedFilter, setdateUploadedFilter] = useState<number>()
+  const [licensesFilter, setLicensesFilter] = useState<number[]>()
+  const [videoLegnthFilter, setVideoLegnthFilter] = useState<VideoLengthOptions>()
+  const [paidPromotionalMaterialFilter, setPaidPromotionalMaterialFilter] = useState<boolean>()
+  const [matureContentRatingFilter, setMatureContentRatingFilter] = useState<boolean>()
```

Comment:
Why don't we put this state directly in the component? I think FiltersBar could manage its own state? if we do that, then all the filters don't have to pass through the consumer component

Reply:
What other fields does the parent need other than `canClearFilters`? If it's only this, we could just pass a single field to the parent instead of all the state

Reply:
Both language and sortBy would be displayed outside of the `FiltersBar` though right? In that case I guess those could be handled by the consumer directly? In that case, only resulting filters and `canClearFilters` would need to be passed to the consumer?